diff -rupN hawkey_old/subject.h hawkey/subject.h
--- hawkey_old/subject.h	2015-01-28 12:43:32.039078599 +0100
+++ hawkey/subject.h	2015-01-28 12:50:57.367066601 +0100
@@ -32,7 +32,11 @@
 extern "C" {
 #endif
 
+#ifdef __cplusplus
+enum _HyForm :short {
+#else
 enum _HyForm {
+#endif
     HY_FORM_NEVRA = 1,
     HY_FORM_NEVR,
     HY_FORM_NEV,
diff -rupN hawkey_old/types.h hawkey/types.h
--- hawkey_old/types.h	2015-01-28 12:43:32.038078594 +0100
+++ hawkey/types.h	2015-01-28 12:55:35.517217260 +0100
@@ -25,7 +25,12 @@
 extern "C" {
 #endif
 
+#ifdef __cplusplus
+enum _HyForm :short ;
+#else
 typedef enum _HyForm HyForm;
+#endif
+
 typedef struct _HyAdvisory * HyAdvisory;
 typedef struct _HyAdvisoryList * HyAdvisoryList;
 typedef struct _HyAdvisoryPkg * HyAdvisoryPkg;
